### PR TITLE
Junit Delete, Add and deleteLine

### LINKED ISSUE #60: Create JUnit tests for Add, Delete

### PR DESCRIPTION
**Pull request by Nazryl**
Closes #60

### CODE REVIEW

**Comment by Nazryl** [comment]
Created PR first. Not ready to be reviewed. Will notify.

File: src/test/java/seedu/duke/command/DeleteNoteCommandTest.java [hunk 1]
```diff
@@ -1,10 +1,145 @@
...
+        DeleteNoteCommand command = new DeleteNoteCommand(targetVisibleIndex);
+        command.setData(notebook, timetable, tagManager, storageManager);
+
+        return command;
+    }
+
+    /**
+     * Creates a new delete command.
+     *
+     * @param targetVisibleString of the person that we want to delete
```
Comment (R-Ramana) [inline src/test/java/seedu/duke/command/DeleteNoteCommandTest.java:113]: Same as above

File: src/test/java/seedu/duke/command/DeleteNoteCommandTest.java [hunk 2]
```diff
@@ -1,10 +1,145 @@
...
+    @Test
+    public void execute_emptyAddressBook_returnsPersonNotFoundMessage() {
+        assertDeletionFailsDueToNoSuchPersonInteger(2, notebook);
+        assertDeletionFailsDueToNoSuchPersonString("ZZZ", notebook);
+    }
+
+    /**
+     * Creates a new delete command.
+     *
+     * @param targetVisibleIndex of the person that we want to delete
```
Comment (R-Ramana) [inline src/test/java/seedu/duke/command/DeleteNoteCommandTest.java:100]: Think you can make this statement clearer, and missing param

File: src/test/java/seedu/duke/command/DeleteNoteCommandTest.java [hunk 3]
```diff
@@ -1,10 +1,145 @@
...
+    private DeleteNoteCommand createDeleteCommand(String targetVisibleString, Notebook notebook) {
+        DeleteNoteCommand command = new DeleteNoteCommand(targetVisibleString);
+        command.setData(notebook, timetable, tagManager, storageManager);
+
+        return command;
+    }
+
+    /**
+     * Asserts that the person at the specified index cannot be deleted, because that person
+     * is not in the address book.
```
Comment (R-Ramana) [inline src/test/java/seedu/duke/command/DeleteNoteCommandTest.java:124]: Add @param for the javaDoc?

File: src/test/java/seedu/duke/command/DeleteNoteCommandTest.java [hunk 4]
```diff
@@ -1,10 +1,145 @@
...
+        DeleteNoteCommand command = createDeleteCommand(visibleIndex, notebook);
+        String result = command.execute();
+
+        assertEquals(DeleteNoteCommand.COMMAND_UNSUCCESSFUL_MESSAGE, result);
+        assertEquals(expectedNotebook.getNotes(), notebook.getNotes());
+    }
+
+    /**
+     * Asserts that the person at the specified index cannot be deleted, because that person
+     * is not in the address book.
```
Comment (R-Ramana) [inline src/test/java/seedu/duke/command/DeleteNoteCommandTest.java:136]: Add @param for the javaDoc?

File: src/test/java/seedu/duke/command/DeleteNoteCommandTest.java [hunk 5]
```diff
@@ -1,10 +1,145 @@
...
+        DeleteNoteCommand command = createDeleteCommand(targetVisibleIndex, notebook);
+        String result = command.execute();
+
+        assertEquals(expectedMessage, result);
+        assertEquals(deletedNotebook.getNotes(), notebook.getNotes());
+    }
+
+    /**
+     * Asserts that the person at the specified index can be successfully deleted.
+     * The addressBook passed in will not be modified (no side effects).
```
Comment (R-Ramana) [inline src/test/java/seedu/duke/command/DeleteNoteCommandTest.java:74]: Add @param for the javaDoc?

File: src/test/java/seedu/duke/command/DeleteNoteCommandTest.java [hunk 6]
```diff
@@ -1,10 +1,145 @@
...
     @Test
-    void execute() {
+    public void execute_validIndex_personIsDeleted() {
+        assertDeletionSuccessfulInteger(1, notebook);
+        assertDeletionSuccessfulString("java", notebook);
+    }
+
+    /**
+     * Asserts that the person at the specified index can be successfully deleted.
+     * The addressBook passed in will not be modified (no side effects).
```
Comment (R-Ramana) [inline src/test/java/seedu/duke/command/DeleteNoteCommandTest.java:58]: Add @param for the javaDoc?

**Review by R-Ramana — APPROVED** [review]

**Review by Chongjx — APPROVED** [review]
Looks good overall

File: src/test/java/seedu/duke/command/DeleteNoteCommandTest.java (hunk 4 above)
Comment (Nazryl) [reply src/test/java/seedu/duke/command/DeleteNoteCommandTest.java:136]: isnt final.

**Pull request closed without merging**